libtsuba/include/katana/EntityTypeManager.h

```diff
@@ -527,6 +527,30 @@ class KATANA_EXPORT EntityTypeManager {
     return std::nullopt;
   }
 
+  /// \returns the names of the atomic supertypes for the given non-atomic type
+  /// \p entity_type_id is an non-atomic type
+  std::set<std::string> GetNonAtomicTypeNames(
+      EntityTypeID entity_type_id) const {
+    std::set<std::string> result;
+    auto atomic_types = GetAtomicSubtypes(entity_type_id);
+    for (EntityTypeID super_type_id = 0; super_type_id < atomic_types.size();
+         ++super_type_id) {
+      if (atomic_types.test(super_type_id)) {
+        auto val = GetAtomicTypeName(super_type_id);
+        if (val) {
+          result.insert(val.value());
+        } else {
+          // Seems it's possible to have non-atomic from another non-atomic.
+          auto atomic_type_names = GetNonAtomicTypeNames(super_type_id);
```

Comment:
@roshandathathri this is why i was asking on property-graph-2 channel. If I understood correctly I have to recursively check. I did a test to confirm this and it seems it's possible. (see [test_property_graph.py](https://github.com/KatanaGraph/katana/pull/976/files#diff-ca79ca88ad2a58b4d04d59925b6352ab90930da91d5bd171f073d1662f12db01) changes)

Just want make sure I didn't misinterpret the meaning of your reply.

Reply:
If this recursive traversal is needed then that should be handled in `GetAtomicSubtypes`. Otherwise it's not actually correct. This function should definitely not have to do anything other than getting the names.

Reply:
I taught about that too but the original method expects a atomic type. a non atomic type would return a null in the code. Now i can change that but I am not sure how other code trace could be affected by this change. that was my only fear

Reply:
@roshandathathri I been thinking about what you said and I want to make sure I am not confused by my understanding. Can you confirm if my test case is even a sensible thing?

Reply:
fixed based on meeting dicussions